src/fromager/bootstrapper.py

```diff
@@ -259,12 +259,30 @@ def bootstrap(self, req: Requirement, req_type: RequirementType) -> Version:
             wheel_filename=wheel_filename,
         )
 
-        if build_sdist_only:
+        if pbi.pre_built:
```

Comment:
I wonder if we could do something simpler here and just switch based on whether we have a wheel filename or not? By the time we get here we've either built the wheel or the sdist based on these other flags, so we just need to know which file to use to pull the dependencies, right? And there are really only 2 options.